pkg/evetestkit/utils.go

```diff
@@ -728,3 +731,121 @@ func InitilizeTest(projectName string, options ...TestOption) (*EveNode, error)
 
 	return rnode, nil
 }
+
+func NewTestContextFromConfig(cfg *openevec.EdenSetupArgs) (*testcontext.TestContext, error) {
+	var (
+		err       error
+		sdnClient *edensdn.SdnClient
+		withSdn   bool
+	)
+
+	devModel := cfg.Eve.DevModel
+	eveRemote := cfg.Eve.Remote
+	withSdn = !cfg.Sdn.Disable &&
+		devModel == defaults.DefaultQemuModel &&
+		!eveRemote
+	if withSdn {
+		sdnSSHPort := cfg.Sdn.SSHPort
+		sdnMgmtPort := cfg.Sdn.MgmtPort
+		sdnSourceDir := filepath.Join(cfg.Eden.Root, strings.TrimSpace(cfg.Sdn.SourceDir))
+		sdnSSHKeyPath := filepath.Join(sdnSourceDir, "vm/cert/ssh/id_rsa")
+		sdnClient = &edensdn.SdnClient{
+			SSHPort:    uint16(sdnSSHPort),
+			SSHKeyPath: sdnSSHKeyPath,
+			MgmtPort:   uint16(sdnMgmtPort),
+		}
+	}
+
+	vars, err := openevec.InitVarsFromConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	ctx := &controller.CloudCtx{Controller: &adam.Ctx{}}
+	ctx.SetVars(vars)
+	if err := ctx.InitWithVars(vars); err != nil {
+		return nil, err
+	}
+	ctx.GetAllNodes()
+	tstCtx := &testcontext.TestContext{
+		Cloud:     ctx,
+		Tests:     map[*device.Ctx]*testing.T{},
+		SdnClient: sdnClient,
+		WithSdn:   withSdn,
+	}
+	tstCtx.ProcBus = testcontext.InitBus(tstCtx)
+	return tstCtx, nil
+
+}
+
+func InitilizeTestFromConfig(projectName string, cfg *openevec.EdenSetupArgs, options ...TestOption) (*EveNode, error) {
```

Comment:
can this be used to initialize tests in `lim_test.go` and `reboot_test.go`?

Reply:
yup, this should be used in all "NeoEden" tests

Reply:
I'd submit a separate PR with that, this one is already big